src/main/java/com/sparrowwallet/sparrow/io/satochip/SatoCardApi.java

```diff
@@ -0,0 +1,514 @@
+import com.sparrowwallet.drongo.crypto.ECDSASignature;
+import com.sparrowwallet.drongo.crypto.SchnorrSignature;
+import com.sparrowwallet.drongo.protocol.*;
+import com.sparrowwallet.drongo.psbt.PSBT;
+import com.sparrowwallet.drongo.psbt.PSBTInput;
+import com.sparrowwallet.drongo.psbt.PSBTInputSigner;
+import com.sparrowwallet.drongo.wallet.*;
+import com.sparrowwallet.sparrow.control.CardImportPane;
+import com.sparrowwallet.sparrow.io.CardApi;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.smartcardio.*;
+import java.util.*;
+import java.nio.charset.StandardCharsets;
+
+public class SatoCardApi extends CardApi {
+    private static final Logger log = LoggerFactory.getLogger(SatoCardApi.class);
+
+    private final WalletModel cardType;
+    private SatochipCommandSet cardProtocol;
+    private String pin;
+    private String basePath = null; // = "m/84'/1'/0'"
+
+    public SatoCardApi(WalletModel cardType, String pin) throws CardException {
+        log.debug("SATOCHIP SatoCardApi() cardType: " + cardType);
+        this.cardType = cardType;
+        this.cardProtocol = new SatochipCommandSet();
+        this.pin = pin;
+    }
+
+    @Override
+    public boolean isInitialized() throws CardException {
+        log.debug("SATOCHIP SatoCardApi isInitialized() START");
+        SatoCardStatus cardStatus = this.getStatus();
+        return cardStatus.isInitialized(); // setupDone && isSeeded
+    }
+
+    //TODO
+    @Override
+    public void initialize(int slot, byte[] seedBytes) throws CardException {
+        log.debug("SATOCHIP SatoCardApi initialize() START");
+        // TODO check device certificate
+        SatoCardStatus cardStatus = this.getStatus();
+
+        APDUResponse rapdu;
+        if (!cardStatus.isSetupDone()){
+            byte maxPinTries = 5;
+            rapdu  = this.cardProtocol.cardSetup(maxPinTries, pin.getBytes(StandardCharsets.UTF_8));
+            // check ok
+        }
+        if (!cardStatus.isSeeded()){
+            // check pin
+            rapdu = this.cardProtocol.cardVerifyPIN(0, pin);
+            // todo: check PIN response
+
+            rapdu = this.cardProtocol.cardBip32ImportSeed(seedBytes);
+            // check ok
+        }
+    }
+
+    @Override
+    public WalletModel getCardType() throws CardException {
+        return WalletModel.SATOCHIP;
+    }
+
+    //TODO
+    @Override
+    public int getCurrentSlot() throws CardException {
+        throw new CardException("Satochip does not support 'getCurrentSlot' !");
+    }
+
+    //TODO
+    @Override
+    public ScriptType getDefaultScriptType() {
+        return ScriptType.P2WPKH;
+    }
+
+    SatoCardStatus getStatus() throws CardException {
```

Comment:
Is package visibility wanted? You could also avoid using a variable and directly return the value, and also remove the `throws CardException` from the method signature